generator/mavgen_java.py

```diff
@@ -58,50 +56,55 @@ def generate_CRC(directory, xml):
  * java mavlink generator tool. It should not be modified by hand.
  */
 
-package com.MAVLink.${basename};
+package com.mavlink.${basename};
+
+import java.util.HashMap;
+import java.util.Map;
 
 /**
-* X.25 CRC calculation for MAVlink messages. The checksum must be initialized,
-* updated with witch field of the message, and then finished with the message
-* id.
-*
-*/
+ * X.25 CRC calculation for MAVlink messages. The checksum must be initialized,
+ * updated with witch field of the message, and then finished with the message
+ * id.
+ */
 public class CRC {
-    private static final int[] MAVLINK_MESSAGE_CRCS = {${message_crcs_array}};
+    private static final Map<Integer, Integer> MAVLINK_MESSAGE_CRCS;
     private static final int CRC_INIT_VALUE = 0xffff;
     private int crcValue;
 
+    static {
+        MAVLINK_MESSAGE_CRCS = new HashMap<>();
+        ${message_crcs_array}
+    }
+
     /**
-    * Accumulate the X.25 CRC by adding one char at a time.
-    *
-    * The checksum function adds the hash of one char at a time to the 16 bit
-    * checksum (uint16_t).
-    *
-    * @param data
-    *            new char to hash
-    **/
-    public  void update_checksum(int data) {
+     * Accumulate the X.25 CRC by adding one char at a time.
+     *
+     * The checksum function adds the hash of one char at a time to the 16 bit
+     * checksum (uint16_t).
+     *
+     * @param data new char to hash
+     **/
+    public void update_checksum(int data) {
         data = data & 0xff; //cast because we want an unsigned type
         int tmp = data ^ (crcValue & 0xff);
         tmp ^= (tmp << 4) & 0xff;
         crcValue = ((crcValue >> 8) & 0xff) ^ (tmp << 8) ^ (tmp << 3) ^ ((tmp >> 4) & 0xf);
     }
 
     /**
-    * Finish the CRC calculation of a message, by running the CRC with the
-    * Magic Byte. This Magic byte has been defined in MAVlink v1.0.
-    *
-    * @param msgid
-    *            The message id number
-    */
+     * Finish the CRC calculation of a message, by running the CRC with the
+     * Magic Byte.
+     *
+     * @param msgid The message id number
+     */
     public void finish_checksum(int msgid) {
-        update_checksum(MAVLINK_MESSAGE_CRCS[msgid]);
+        if(MAVLINK_MESSAGE_CRCS.containsKey(msgid))
```

Comment:
I don't see how this was resolved?

Reply:
I think this is another case where we're attempting to duplicate the C mavlink_helpers logic. You can see [here](https://github.com/ArduPilot/pymavlink/blob/33bc90ef2df11d99cb1898a852096f16535fa591/generator/C/include_v2.0/mavlink_helpers.h#L727) if there is no msg_info, then 0 is added to the checksum.

I can modify so that a 0 is accumulated into the checksum when the message_crc is not found, making this functionality identical with the C mavlink_helper? Otherwise this should probably be handled in a separate PR so all the helper methods have consistent behavior.

Reply:
My point is more that we should be throwing an exception or similar here, rather then silently ingesting bad data. It should be the case if we hit this we will probably fail out later, but it's not immediately clear to me that we won't hand back an invalid packet in this case.